Scripts/Proxies/ProxyAnimator.cs

```diff
@@ -27,26 +30,36 @@ public void Apply(Transform transform)
     [SerializeField]
     ActionMap m_ProxyActionMap;
 
-    ProxyHelper.ButtonObject[] m_Buttons;
+    [SerializeField]
+    ProxyAffordanceMap m_AffordanceMap;
+
+    Affordance[] m_Affordances;
+    ProxyAffordanceMap.AffordanceDefinition[] m_AffordanceDefinitions;
     InputControl[] m_Controls;
 
     readonly Dictionary<Transform, TransformInfo> m_TransformInfos = new Dictionary<Transform, TransformInfo>();
 
+    bool m_RightHandedProxy;
+    Node m_Node;
+
     public ActionMap actionMap { get { return m_ProxyActionMap; } }
     public bool ignoreLocking { get { return true; } }
-    internal event Action<ProxyHelper.ButtonObject[], Dictionary<Transform, TransformInfo>, ActionMapInput> postAnimate;
+    public Node node { get { return m_Node; } set { m_Node = value; m_RightHandedProxy = m_Node == Node.RightHand; } }
 
-    void Start()
+    internal event Action<Affordance[], AffordanceDefinition[], Dictionary<Transform, TransformInfo>, ActionMapInput> postAnimate;
+
+    public void Setup(Affordance[] affordances)
     {
-        m_Buttons = GetComponent<ProxyHelper>().buttons;
+        m_Affordances = affordances;
+        m_AffordanceDefinitions = m_AffordanceMap.AffordanceDefinitions;
```

Comment:
Can this second part happen in Awake or Start? It may seem excessive to add another method, but I don't want to give the impression that this somehow relies on m_Affordances being set.